src/main/kotlin/com/github/red/codenames/domain/ports/BoardService.kt

```diff
@@ -37,18 +37,41 @@ class BoardService(private val cardRepository: CardRepository,
         return boardRepository.save(board)
     }
 
-    fun listBoards(): List<Board> = boardRepository.findAll()
-
-    fun clickCard(boardId: String, cardId: Int): Board {
+    fun clickCard(boardId: String, cardId: Int): BoardClickResult? =
         getBoard(boardId)?.let {
-            it.fields[cardId] = it.fields[cardId].copy(clicked = true);
-            boardRepository.save(it)
+            val card = it.fields[cardId]
+            if (card.clicked) BoardClickResult(it, null)
+            else {
+                val clickedCard = card.copy(clicked = true)
+                val winner = when (clickedCard.type) {
+                    Type.ASSASIN -> otherTeam(it.starts)
+                    Type.PASSERBY -> null
+                    Type.RED -> if (allSolved(it, Type.RED)) Team.RED else null
+                    Type.BLUE -> if (allSolved(it, Type.BLUE)) Team.BLUE else null
+                }
+                val nextStarts = when (clickedCard.type) {
```

Comment:
jesli tak rozwiazujesz czyj ruch to trzeba jeszcze zrobic jakies przechodzenie na druga druzyne gdy wszystkie swoje szanse na odkrywanie sie zuzyje